### PR TITLE
Jetpack Thank You Card: Fix Jetpack version check comparison and error detection

### PR DESCRIPTION
**Pull request by oskosk**
#### Changes introduced in this PR

* Verifies that we have a version before comparing it to the minimum Jetpack version required
* FIxes case in "Visit your site" button.
* Makes the `isErrored()` method return true for plugins that falied only when their status is not `'done'`.

#### Testing instructions

1. Go to the staging environment for Calypso
1. Disconnect your site
1. Reconnect
1. Acquire a paid plan.
1. Expect to see the new Thank You Page card

### CODE REVIEW

File: client/my-sites/checkout/checkout-thank-you/jetpack-thank-you-card.jsx [hunk 1]
```diff
@@ -642,7 +642,7 @@ class JetpackThankYouCard extends Component {
 					<a
 						className={ classNames( 'button', 'thank-you-card__button', { 'is-placeholder': ! buttonUrl } ) }
 						href={ buttonUrl }>
-						{ translate( 'Visit Your Site' ) }
+						{ translate( 'Visit your site' ) }
```
Comment (a8ci18n) [inline client/my-sites/checkout/checkout-thank-you/jetpack-thank-you-card.jsx]: :disappointed: 38 existing translations will be lost with this change.

Comment (oskosk) [reply client/my-sites/checkout/checkout-thank-you/jetpack-thank-you-card.jsx]: I know :(

File: client/my-sites/checkout/checkout-thank-you/jetpack-thank-you-card.jsx [hunk 2]
```diff
@@ -353,7 +355,7 @@ class JetpackThankYouCard extends Component {
 		if ( reasons && reasons.length > 0 ) {
 			reason = translate( 'We can\'t modify files on your site.' );
 			this.trackConfigFinished( 'calypso_plans_autoconfig_error_filemod', { error: reason } );
-		} else if ( ! selectedSite.hasMinimumJetpackVersion ) {
+		} else if ( selectedSite.hasMinimumJetpackVersion === false ) {
```
Comment (johnHackworth) [inline client/my-sites/checkout/checkout-thank-you/jetpack-thank-you-card.jsx]: Hmm, I don't see how this change can help here... as `hasMiniumJetpackVersion` is computed, if there's no jetpack version number synced yet, its value is going to be false anyway

Comment (oskosk) [reply client/my-sites/checkout/checkout-thank-you/jetpack-thank-you-card.jsx]: The value would be `null` if the don't have the number version yet. 

https://github.com/Automattic/wp-calypso/blob/master/client/state/sites/selectors.js#L940

Comment (johnHackworth) [reply client/my-sites/checkout/checkout-thank-you/jetpack-thank-you-card.jsx]: ooooooh, cr*p!! I was looking at the old computed version: 

https://github.com/Automattic/wp-calypso/blob/master/client/lib/site/jetpack.js#L25

good enough! :D

**Comment by johnHackworth** [comment]
ok, looks good! :+1:

**Pull request merged**